KyberElasticAdapter/contracts/BaseAdapter.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC20.sol";
+import "./interfaces/IWETH.sol";
+import "./lib/SafeERC20.sol";
+import "./lib/Maintainable.sol";
+
+abstract contract BaseAdapter is Maintainable {
+    using SafeERC20 for IERC20;
+
+    event BaseAdapterSwap(
+        address indexed _tokenFrom,
+        address indexed _tokenTo,
+        uint256 _amountIn,
+        uint256 _amountOut
+    );
+    event UpdatedGasEstimate(address indexed _adapter, uint256 _newEstimate);
+    event Recovered(address indexed _asset, uint256 amount);
+
+    uint256 internal constant UINT_MAX = type(uint256).max;
+    uint256 public swapGasEstimate;
+    string public name;
+
+    constructor(string memory _name, uint256 _gasEstimate) {
+        setName(_name);
+        setSwapGasEstimate(_gasEstimate);
+    }
+
+    function setName(string memory _name) internal {
+        require(bytes(_name).length != 0, "Invalid adapter name");
+        name = _name;
+    }
+
+    function setSwapGasEstimate(uint256 _estimate) public onlyMaintainer {
+        require(_estimate != 0, "Invalid gas-estimate");
+        swapGasEstimate = _estimate;
+        emit UpdatedGasEstimate(address(this), _estimate);
+    }
+
+    function revokeAllowance(address _token, address _spender)
+        external
+        onlyMaintainer
+    {
+        IERC20(_token).safeApprove(_spender, 0);
+    }
+
+    function recoverERC20(address _tokenAddress, uint256 _tokenAmount)
+        external
+        onlyMaintainer
+    {
+        require(_tokenAmount > 0, "BaseAdapter: Nothing to recover");
+        IERC20(_tokenAddress).safeTransfer(msg.sender, _tokenAmount);
+        emit Recovered(_tokenAddress, _tokenAmount);
+    }
+
+    function recoverETH(uint256 _amount) external onlyMaintainer {
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html